test/lib/ansible_test/_data/injector/virtualenv-isolated.sh

```diff
@@ -14,5 +14,5 @@ source "${OUTPUT_DIR}/venv/bin/activate"
 set -ux
 
 if [[ "${ANSIBLE_TEST_COVERAGE}" ]]; then
-    pip install coverage -c ../../../runner/requirements/constraints.txt --disable-pip-version-check
+    pip install coverage -c "$OUTPUT_DIR/../../../lib/ansible_test/_data/requirements/constraints.txt" --disable-pip-version-check
```

Comment:
This path isn't correct for collections. Try this instead:

```suggestion
    pip install coverage -c "${_ANSIBLE_COVERAGE_CONSTRAINTS}" --disable-pip-version-check
```

Then set that variable here:

https://github.com/ansible/ansible/blob/c7473828c7f9ce817d30e00730751e50269bfba6/test/lib/ansible_test/_internal/executor.py#L1597-L1604

The leading underscore in the variable name is used to indicate the variable should not be used in tests.

That's important, since the coverage constraints file may not be used in the future. We don't want tests depending on it directly.